api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package piccolo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/model"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/contexts"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability"
+	observ_utils "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability/utils"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/logger"
+)
+
+const loggerName = "providers.target.piccolo"
+
+var sLog = logger.NewLogger(loggerName)
+
+type PiccoloTargetProviderConfig struct {
+	Name string `json:"name"`
+	Url  string `json:"url"`
+}
+
+type PiccoloTargetProvider struct {
+	Config  PiccoloTargetProviderConfig
+	Context *contexts.ManagerContext
+}
+
+func PiccoloTargetProviderConfigFromMap(properties map[string]string) (PiccoloTargetProviderConfig, error) {
+	ret := PiccoloTargetProviderConfig{}
+	if v, ok := properties["name"]; ok {
+		ret.Name = v
+	}
```

Comment:
Extract url from properties here. Throw COAError if url is not found in properties.